Library/OcCryptoLib/OcCryptoLib.inf

```diff
@@ -47,6 +48,8 @@
 
 [Sources.X64]
   X64/BigNumWordMul64.c
+  X64/Sha2-avx.c
```

Comment:
Similar changes here. You will also need to provide a dummy implementation for Ia32, so that while it never supports AVX, it always compiles without issues.

I.e. have all your functions with `ASSERT (FALSE);` contents and return `FALSE` from the AVX detection function.